apps/documents/assets/ParagraphForm.jsx

```diff
@@ -1,80 +1,74 @@
 var React = require('react')
 var django = require('django')
+var ErrorList = require('../../contrib/static/js/ErrorList')
 
-var ckGet = function (id) {
+const ckGet = function (id) {
   return window.CKEDITOR.instances[id]
 }
 
-var ckReplace = function (id, config) {
+const ckReplace = function (id, config) {
   return window.CKEDITOR.replace(id, config)
 }
 
-var Paragraph = React.createClass({
-  add: function () {
-    this.props.addParagraphBeforeIndex(this.props.index)
-  },
-  delete: function () {
-    this.props.deleteParagraph(this.props.index)
-  },
-  up: function () {
-    this.props.moveParagraphUp(this.props.index)
+const Paragraph = React.createClass({
+
+  handleNameChange: function (e) {
+    const name = e.target.value
+    this.props.onNameChange(name)
   },
-  down: function () {
-    this.props.moveParagraphDown(this.props.index)
+
+  handleTextChange: function (text) {
+    this.props.onTextChange(text)
   },
-  handleNameChange: function (e) {
-    var index = this.props.index
-    var text = e.target.value
-    this.props.updateParagraphName(index, text)
+
+  ckId: function () {
+    return 'id_paragraphs-' + this.props.id + '-text'
   },
+
   ckEditorDestroy: function () {
-    var id = 'id_paragraphs-' + this.props.id + '-text'
-    var editor = ckGet(id)
+    const editor = ckGet(this.ckId())
     if (editor) {
       editor.destroy()
     }
   },
+
   ckEditorCreate: function () {
-    var id = 'id_paragraphs-' + this.props.id + '-text'
-    if (!ckGet(id)) {
-      var editor = ckReplace(id, this.props.config)
+    if (!ckGet(this.ckId())) {
+      var editor = ckReplace(this.ckId(), this.props.config)
       editor.on('change', function (e) {
         var text = e.editor.getData()
-        var index = this.props.index
-        this.props.updateParagraphText(index, text)
+        this.handleTextChange(text)
       }.bind(this))
       editor.setData(this.props.paragraph.text)
     }
   },
+
   componentWillUpdate: function (nextProps) {
     if (nextProps.index > this.props.index) {
-      this.ckEditorDestroy()
+      this.ckEditorDestroy() // why is this needed?
```

Comment:
see a4fc734de131105d2f194bac680410d0ad79586e: We had the issue that ckeditor became unresponsive after moving it around. I do not remember whether this was because of react or only in combination with FlipMove.